testeng/jobs/upgradePythonRequirements.groovy

```diff
@@ -643,7 +650,7 @@ jobConfigs.each { jobConfig ->
                     credentials('jenkins-worker')
                     url("git@github.com:${jobConfig.org}/${jobConfig.repoName}.git")
                 }
-                branch("${jobConfig.targetBranch}")
+                branch(targetBranch)
```

Comment:
I recall this being the part that didn't work as intended last time; you may want to model after these other jobs that check out a branch based on a parameter:

* https://github.com/edx/jenkins-job-dsl/blob/master/platform/jobs/edxEndToEndTestsJob.groovy#L186
* https://github.com/edx/jenkins-job-dsl/blob/master/platform/jobs/RetirementJobs.groovy#L152
* https://github.com/edx/jenkins-job-dsl/blob/master/testingPipelines/jobs/generalPRPipeline.groovy#L146